plugins/code-coverage/src/components/FileExplorer/FileExplorer.tsx

```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useApi } from '@backstage/core-api';
+import { useEntity } from '@backstage/plugin-catalog-react';
+import { Progress, Table, TableColumn } from '@backstage/core';
+import {
+  Box,
+  Card,
+  CardContent,
+  Modal,
+  Tooltip,
+  Typography,
+} from '@material-ui/core';
+import React, { Fragment, useEffect, useState } from 'react';
+import { useAsync } from 'react-use';
+import { codeCoverageApiRef } from '../../api';
+import { Alert } from '@material-ui/lab';
+import DescriptionIcon from '@material-ui/icons/Description';
+import { FileContent } from './FileContent';
+
+export type FileCoverage = {
+  filename: string;
+  branchHits: { [branch: number]: number };
+  lineHits: { [line: number]: number };
+};
+
+type FileStructureObject = Record<string, any>;
+
+type CoverageTableRow = {
+  filename?: string;
+  files: CoverageTableRow[];
+  coverage: number;
+  missing: number;
+  tracked: number;
+  path: string;
+  tableData?: { id: number };
+};
+
+const buildFileStructure = (row: CoverageTableRow) => {
+  const dataGroupedByPath: FileStructureObject = row.files.reduce(
+    (acc: FileStructureObject, cur: CoverageTableRow) => {
+      let path = cur.filename;
+      if (row.path) {
+        path = path?.split(`${row.path}/`)[1];
```

Comment:
hm would `path?.substring(row.path.length + 1)` be safer?

Reply:
not sure if that'd work... @nissayeva wdyt?